src/chocolatey/infrastructure/information/ProcessInformation.cs

```diff
@@ -158,6 +165,130 @@ public static bool UserIsSystem()
             return isSystem;
         }
 
+        public static ProcessTree GetProcessTree()
+        {
+            return GetProcessTree(null);
+        }
+
+        public static ProcessTree GetProcessTree(Process process)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            var tree = new ProcessTree(process.ProcessName);
+
+            if (Platform.GetPlatform() != PlatformType.Windows)
+            {
+                return tree;
+            }
+
+            try
+            {
+                try
+                {
+                    tree = PopulateProcessTreeInternal(tree, process);
+                }
+                catch (TypeLoadException ex) when (ex is DllNotFoundException || ex is EntryPointNotFoundException)
+                {
+                    try
+                    {
+                        // These exceptions mean the lookup failed because the DLL is missing or the entry point is no longer present.
+                        // Ignore these and fall back to the alternative p/invoke method if we haven't already.
+                        tree = PopulateProcessTreeStable(tree, process);
+                    }
+                    catch (TypeLoadException)
+                    {
+                        "chocolatey".Log().Warn(logging.ChocolateyLoggers.LogFileOnly, "All available methods of querying processes from the win32 APIs are broken or critical DLLs are missing.");
+                    }
+                }
+            }
+            catch (Win32Exception ex)
+            {
+                "chocolatey".Log().Warn(logging.ChocolateyLoggers.LogFileOnly, "Unhandled Win32Exception ({0}) in finding parent processes.", ex.Message);

```

Comment:
I'm not... sure what you're getting at here? What would be special about that case that we would need to handle this differently?

This has nothing intrinsically different in that command as far as I know? All this affects is whether we can send a user-agent string with the process tree info, or whether that portion of the user agent remains blank.

I don't think this is really intended to be something the user has to worry about -- either we can get the process tree or we can't; the user can't _do_ anything about that either way.

Reply:
This is outputted to the log file only, and is a warning on purpose.

As Rain said, it is not something for the user, but rather for us. The reason it is a warning is that it will be easier to notice by either us, or support when/if this will happen.

Reply:
Yeah, I think I saw `Warn` and just went immediately to it being shown to the user. I see now that it's log file only.